kimchi/src/circuits/expr.rs

```diff
@@ -2146,15 +2146,54 @@ where
 pub mod constraints {
     use super::*;
 
+    /// This trait defines a common arithmetic operations interface
+    /// that can be used by constraints.  It allows us to reuse
+    /// constraint code for witness computation.
+    pub trait ArithmeticOps:
+        std::ops::Add<Output = Self>
+        + std::ops::Sub<Output = Self>
+        + std::ops::Neg<Output = Self>
+        + std::ops::Mul<Output = Self>
+        + Clone
+        + Zero
+        + One
+        + From<u64>
+    // Add more as necessary
```

Comment:
just to know, is there an specific reason to no have Copy too? like a limitation in some implementor.

Reply:
No reason that I'm aware of.  I just added the minimal set that the current codebase requires 👍🏼